eta/core/video.py

```diff
@@ -946,6 +962,47 @@ def _validate_schema(self):
             for frame_labels in itervalues(self.frames):
                 self._validate_frame_labels(frame_labels)
 
+    def _filter_by_schema(self):
+        if self.has_schema:
+            self._filter_video_attributes()
+            for frame_key, frame_labels in iteritems(self.frames):
+                self.frames[frame_key] = self._filter_frame_labels(frame_labels)
+
+    def _filter_video_attributes(self):
```

Comment:
FYI- `eta.core.serial.Container` has a builtin function for filtering. You could write this as:

```py
def _filter_video_attributes(self):
    filter_fcn = lambda attr: self.schema.has_video_attribute(attr.name)
    self.attrs = sellf.attrs.get_matches([filter_fcn])
```

That said, I would understand if you didn't want to write it like that. A loop can be more clear